pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/KitUploadRoute.java

```diff
@@ -404,8 +403,15 @@ private void addKitRequest(Connection conn, String kitTypeName, KitRequestSettin
 
             //If there is a participant change the participantID to the ID of the existing
             //participant
-            if (StringUtils.isNotBlank((ddpInstance.getParticipantIndexES()))) {
-                participantID = kit.getParticipantId().trim();
+            if (ddpInstance.hasEsIndex()) {
+                //check if there are previous kits with the legacy participant id
+                String participantIdOfLegacyKit = KitRequestShipping.getParticipantIdFromLegacyKit(ddpInstance, participantID,
```

Comment:
See if legacy kit were uploaded using the altpid and if so, add the new kits with that one instead of ddpParticipantId